test/csuite/wt5212_incremental_backup/main.c

```diff
@@ -0,0 +1,779 @@
+         */
+        printf("Iteration %d: taking incremental backup\n", i);
+        take_incr_backup(session, i);
+
+        printf("Iteration %d: dumping and comparing data\n", i);
+        testutil_check(compare_backups(i, uri));
+    }
+}
+
+/*
+ * Remove all the records, take backups and validate
+ */
+static void
+remove_all_records_validate(WT_SESSION *session)
+{
+    int i, j;
+
+    remove_work(session, 0, 0);
+    for (i = 1; i < ITERATIONS_MULTIPLIER; i++) {
+        printf("Iteration %d: Removing data\n", i);
+        for (j = 0; j < i; j++) {
+            remove_work(session, i, j);
+        }
+    }
+    take_full_backup(session, i);
+
+    take_incr_backup(session, i);
+
+    printf("Dumping and comparing data\n");
+    testutil_check(compare_backups(i, uri));
+}
+
+static void
+drop_old_add_new_objects(WT_CONNECTION *wt_conn)
+{
+    WT_CURSOR *backup_cur;
+    WT_SESSION *session;
+    int ret, i, j;
+    char buf[1024];
+
+    testutil_check(wt_conn->close(wt_conn, NULL));
+    testutil_check(wiredtiger_open(home, NULL, CONN_CONFIG, &wt_conn));
+    testutil_check(wt_conn->open_session(wt_conn, NULL, NULL, &session));
+    testutil_check(session->create(session, uri3, "key_format=S,value_format=S"));
+
+    testutil_check(session->open_cursor(session, "backup:", NULL, NULL, &backup_cur));
+    testutil_check(backup_cur->close(backup_cur));
+
+    testutil_check(session->drop(session, uri2, "force"));
+
+    new_object = true;
+
+    for (i = ITERATIONS_MULTIPLIER + 1; i < ITERATIONS_MULTIPLIER * 2; i++) {
+        printf("Iteration %d: adding data\n", i);
+        for (j = ITERATIONS_MULTIPLIER; j < i; j++) {
+            add_work(session, i, j, uri, uri3);
+            testutil_check(session->checkpoint(session, NULL));
+        }
+
+        printf("Iteration %d: taking incremental backup\n", i);
+        take_incr_backup(session, i);
+
+        // Check if the dropped object exists in the incremental folder.
+        testutil_check(__wt_snprintf(
+          buf, sizeof(buf), "%s -R -h %s.%d list | grep %s", WT_UTIL, home_incr, i, uri2));
+        ret = system(buf);
+        testutil_assertfmt(ret != 0, "%s dropped, but file exists in %s.%d\n", uri2, home_incr, i);
+
+        // Clean up
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "rm -rf %s.%d", home_incr, i));
+        testutil_check(system(buf));
+    }
+    new_object = false;
+}
+
+/*
+ * Create previously dropped object and add new contents to it
+ */
+static void
+create_dropped_object_add_new_content(WT_CONNECTION *wt_conn)
+{
+    WT_SESSION *session;
+    int i, j;
+
+    testutil_check(wt_conn->close(wt_conn, NULL));
+    testutil_check(wiredtiger_open(home, NULL, CONN_CONFIG, &wt_conn));
+    testutil_check(wt_conn->open_session(wt_conn, NULL, NULL, &session));
+    testutil_check(session->create(session, uri2, "key_format=S,value_format=S"));
+
+    for (i = ITERATIONS_MULTIPLIER * 2; i < ITERATIONS_MULTIPLIER * 3; i++) {
```

Comment:
Again, I would think about if the loop is necessary or if one time through would be enough for testing.

Reply:
I have changed the code to run once. Thanks.